services/spar/src/Spar/App.hs

```diff
@@ -159,20 +162,39 @@ getUser uref = do
 -- FUTUREWORK: once we support <https://github.com/wireapp/hscim scim>, brig will refuse to delete
 -- users that have an sso id, unless the request comes from spar.  then we can make users
 -- undeletable in the team admin page, and ask admins to go talk to their IdP system.
-createUser :: SAML.UserRef -> Maybe Name -> ManagedBy -> Spar UserId
-createUser suid mbName managedBy = do
+createSamlUser :: SAML.UserRef -> Maybe Name -> ManagedBy -> Spar UserId
+createSamlUser suid mbName managedBy = do
   buid <- Id <$> liftIO UUID.nextRandom
-  createUser_ buid suid mbName managedBy
+  createSamlUserWithId buid suid mbName managedBy
   pure buid
 
--- | Like 'createUser', but for an already existing 'UserId'.
-createUser_ :: UserId -> SAML.UserRef -> Maybe Name -> ManagedBy -> Spar ()
-createUser_ buid suid mbName managedBy = do
+-- | Like 'createSamlUser', but for an already existing 'UserId'.
+createSamlUserWithId :: UserId -> SAML.UserRef -> Maybe Name -> ManagedBy -> Spar ()
+createSamlUserWithId buid suid mbName managedBy = do
   teamid <- (^. idpExtraInfo) <$> getIdPConfigByIssuer (suid ^. uidTenant)
   insertUser suid buid
   buid' <- Intra.createBrigUser suid buid teamid mbName managedBy
   assert (buid == buid') $ pure ()
 
+-- | If the team has no scim token, call 'createSamlUser'.  Otherwise, raise "invalid
+-- credentials".
+autoprovisionSamlUser :: SAML.UserRef -> Maybe Name -> ManagedBy -> Spar UserId
+autoprovisionSamlUser suid mbName managedBy = do
+  buid <- Id <$> liftIO UUID.nextRandom
+  autoprovisionSamlUserWithId buid suid mbName managedBy
+  pure buid
+
+-- | Like 'autoprovisionSamlUser', but for an already existing 'UserId'.
+autoprovisionSamlUserWithId :: UserId -> SAML.UserRef -> Maybe Name -> ManagedBy -> Spar ()
+autoprovisionSamlUserWithId buid suid mbName managedBy = do
+  teamid <- (^. idpExtraInfo) <$> getIdPConfigByIssuer (suid ^. uidTenant)
+  scimtoks <- wrapMonadClient $ Data.getScimTokens teamid
+  if null scimtoks
+    then createSamlUserWithId buid suid mbName managedBy
+    else throwError . SAML.Forbidden $
+            "bad credentials (note that your team has uses SCIM, " <>
```

Comment:
"bad credentials" might mislead someone to think that they actually have wrong credentials 🤔

I'd say this is a clear-cut case of "bad request".

Reply:
I disagree.  it's arguably an error in the IdP setup because they send people our way that they should have created first, but it's a perfectly valid request for all we know.  Just a user we haven't seen before.

Does this make sense?  If not feel free to rename it in a separate PR.